### PR TITLE
Finished with updates on the package after notebook incorporation

### PR DESCRIPTION
**Pull request by SaarthShah**
## Description

1) The fetch functions work for all 5 devices.
2) Generated synthetic data for all 5 devices, precisely modeled off accurate data.
3) I might still need some docstrings; I would love some reviews on where I can add them to make it easier for the users to know how to use the proper device classes.
4) Yet to generate the unit tests, I need some review of the code before I do that to ensure the fetch and generate code meets all expectations.
5) Fixed minor bugs when incorporating the notebooks with actual data.
6) Fixed implementation of packages and dependencies.

## PS
1) I Could not get the black formatter to work; I used autopep8 for now.
2) Cronometer has a lot of food items for servings for my generation of dummy data. I added random different items into cronometer and then am randomly sampling between those for my synthetic data generation.

### CODE REVIEW

File: wearipedia/devices/cronometer/cronometer_synthetic.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
..., 15: 'Cereal Grains and Pasta', 16: 'Fruits and Fruit Juices', 17: 'Vegetables and Vegetable Products', 18: 'Cereal Grains and Pasta', 19: 'Vegetables and Vegetable Products', 20: 'Fruits and Fruit Juices', 21: 'Fast Foods', 22: 'Fast Foods', 23: 'Vegetables and Vegetable Products', 24: 'Cereal Grains and Pasta', 25: 'Fast Foods', 26: 'Fast Foods', 27: 'Fast Foods', 28: 'Fast Foods', 29: 'Fast Foods', 30: 'Fast Foods', 31: 'Fast Foods', 32: 'Baked Products', 33: 'Fruits and Fruit Juices', 34: 'Meals, Entrees, and Sidedishes', 35: 'Soups, Sauces, and Gravies', 36: 'Meals, Entrees, and Sidedishes', 37: 'Vegetables and Vegetable Products', 38: 'Fruits and Fruit Juices', 39: 'Dairy and Egg Products', 40: 'Fast Foods', 41: 'Meals, Entrees, and Sidedishes', 42: 'Sweets', 43: 'Baked Products', 44: 'Meals, Entrees, and Sidedishes', 45: 'Snacks', 46: 'Cereal Grains and Pasta', 47: 'Meals, Entrees, and Sidedishes', 48: 'Breakfast Cereals', 49: 'Fast Foods', 50: 'Fast Foods', 51: 'Fruits and Fruit Juices', 52: 'Meals, Entrees, and Sidedishes', 53: 'Cereal Grains and Pasta', 54: 'Soups, Sauces, and Gravies', 55: 'Vegetables and Vegetable Products', 56: 'Sweets', 57: 'Dairy and Egg Products'}}
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/cronometer/cronometer_synthetic.py:25]: Alright, let me try something

**Comment by rodrigo-castellon** [comment]
As a heads up, I've made a high-level refactor to all devices, which eliminates `params` as input to the device constructor and replaces it with just a set of kwargs. See the new code (e.g. https://github.com/Stanford-Health/wearipedia/blob/main/wearipedia/devices/garmin/fenix_7s.py) to see what to do.

File: wearipedia/devices/underarmour/myfitnesspal_synthetic.py [hunk 2]
```diff
@@ -0,0 +1,928 @@
...
+            },
+            'totals': {
+                'calories': float(snack_item['calories']),
+                'carbohydrates': float(snack_item['carbohydrates']),
+                'fat': float(snack_item['fat']),
+                'protein': float(snack_item['protein']),
+                'sodium': float(snack_item['sodium']),
+                'sugar': float(snack_item['sugar'])
+            }
+        }])
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/underarmour/myfitnesspal_synthetic.py:926]: Fixed

File: wearipedia/devices/cronometer/cronometer.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    def _default_params(self):
+        return {
+            "start_date": self.init_params["synthetic_start_date"],
+            "end_date": self.init_params["synthetic_end_date"],
+        }
+
+    def _get_real(self, data_type, params):
+        return fetch_real_data(
+            self, params["start_date"], params["end_date"], data_type
+        )
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/cronometer/cronometer.py:47]: Fixed, cronometer they had changed one of the API endpoints. Got it working now

File: wearipedia/devices/strava/strava_syn_gen.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                        coordinate[1]+np.random.normal(-0.005, 0.005)),
+                       (coordinate[0]+np.random.normal(-0.005, 0.005),
+                        coordinate[1]+np.random.normal(-0.005, 0.005)),
+                       (coordinate[0]+np.random.normal(-0.005, 0.005),
+                        coordinate[1]+np.random.normal(-0.005, 0.005)),
+                       (coordinate[0]+np.random.normal(-0.005, 0.005),
+                        coordinate[1]+np.random.normal(-0.005, 0.005)),
+                       (coordinate[0]+np.random.normal(-0.005, 0.005),
+                        coordinate[1]+np.random.normal(-0.005, 0.005)),
+                       (coordinate[0]+np.random.normal(-0.005, 0.005), coordinate[1]+np.random.normal(-0.005, 0.005))]
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/strava/strava_syn_gen.py:103]: done

File: wearipedia/devices/cronometer/cronometer_synthetic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/cronometer/cronometer_synthetic.py:25]: Shifted all the data into csv file instead, much cleaner

File: wearipedia/devices/underarmour/myfitnesspal.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        if 'cookies' in auth_creds:
+            try:
+                client = myfitnesspal.Client(auth_creds['cookies'])
+            except myfitnesspal.exceptions.MyfitnesspalLoginError as e:
+                print(
+                    "Could not authenticate with MyFitnessPal using the cookies provided, retry using a local machine")
+                return
+        else:
+            try:
+                client = myfitnesspal.Client(browsercookie.load())
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/underarmour/myfitnesspal.py:86]: So, the package automatically accesses internal cookies stored in a machine and I have already added code and instructions to pass custom cookies, I don't think we need this.

File: wearipedia/devices/google/googlefit.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        # POST request to get the participant's ACCESS TOKEN
+        res = requests.get(auth_url, headers=headers)
+
+        # saving the participant's ACCESS TOKEN
+        if res.status_code == 200:
+            self.access_token = auth_creds['access_token']
+            print('Authenticated!\n')
+        else:
+            print('Authentication failed!\n')
+            print(res.text)
```
[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/google/googlefit.py:102]: Now, the user can use either the access token or authorization code to login into wearipedia. Does this work?

[the remark this answers is not in the document]
Comment (rodrigo-castellon) [reply wearipedia/devices/google/googlefit.py:102]: I'm not sure if I see the updated code, did you check it in?

[the remark this answers is not in the document]
Comment (SaarthShah) [reply wearipedia/devices/google/googlefit.py:102]: not yet, I am yet to push it

**Comment by SaarthShah** [comment]
Created new pull req

**Pull request closed without merging**